base_layer/core/src/transactions/key_manager/inner.rs

```diff
@@ -1003,26 +1070,92 @@ where TBackend: KeyManagerBackend<PublicKey> + 'static
 
                 #[cfg(feature = "ledger")]
                 {
-                    let mut total_script_private_key = PrivateKey::default();
+                    let mut partial_script_offset = PrivateKey::default();
+                    let mut derived_script_keys = vec![];
+                    let mut script_key_indexes = vec![];
+                    for script_key_id in script_key_ids {
+                        match script_key_id {
+                            TariKeyId::Managed { branch, index } => {
+                                match TransactionKeyManagerBranch::from_key(branch) {
+                                    TransactionKeyManagerBranch::Spend | TransactionKeyManagerBranch::PreMine => {
+                                        script_key_indexes
+                                            .push((TransactionKeyManagerBranch::from_key(branch), *index));
+                                    },
+                                    _ => {
+                                        return Err(TransactionError::from(
+                                            self.branch_not_supported_error("get_script_offset", branch),
+                                        ));
+                                    },
+                                }
+                            },
+                            TariKeyId::Derived { key } => {
+                                // TODO: What about if the derived key is a TariKeyId::Managed, but
+                                // TODO: `self.get_private_key(...)` will error?
+                                let key_id = TariKeyId::from_str(key.to_string().as_str())
+                                    .map_err(|_| KeyManagerServiceError::KeySerializationError)?;
+                                let k = self.get_private_key(&key_id).await?;
+                                derived_script_keys.push(k);
```

Comment:
```suggestion
                                let key_id = TariKeyId::from_str(key.to_string().as_str())
                                    .map_err(|_| KeyManagerServiceError::KeySerializationError)?;
                                let k = self.get_private_key(&key_id).await?;
                                derived_script_keys.push(k);
```
if self.get_private_key(...) errors, then the higher logic assigned the wrong key. Its fine if we return the error form get_private_key